packages/pn-pa-webapp/src/pages/__test__/Dashboard.page.test.tsx

```diff
@@ -105,81 +125,132 @@ describe('Dashboard Page', () => {
     const itemsPerPageSelector = result?.queryByTestId('itemsPerPageSelector');
     expect(itemsPerPageSelector).toBeInTheDocument();
     const pageSelector = result?.queryByTestId('pageSelector');
     expect(pageSelector).toBeInTheDocument();
-    expect(mockDispatchFn).toBeCalledTimes(1);
-    expect(mockActionFn).toBeCalledTimes(1);
-    expect(mockActionFn).toBeCalledWith({
-      startDate: formatToTimezoneString(tenYearsAgo),
-      endDate: formatToTimezoneString(getNextDay(today)),
-      size: 10,
-    });
   });
 
   it('changes items per page', async () => {
+    mock = mockApi(
+      apiClient,
+      'GET',
+      NOTIFICATIONS_LIST({
+        startDate: formatToTimezoneString(tenYearsAgo),
+        endDate: formatToTimezoneString(getNextDay(today)),
+        size: 1,
+      }),
+      200,
+      undefined,
+      notificationsFromBe
+    );
+    mockApi(
+      mock,
+      'GET',
+      NOTIFICATIONS_LIST({
+        startDate: formatToTimezoneString(tenYearsAgo),
+        endDate: formatToTimezoneString(getNextDay(today)),
+        size: 50,
+      }),
+      200,
+      undefined,
+      notificationsFromBe2rows
+    );
     await act(async () => {
-      result = render(<Dashboard />, initialState(notificationsToFe.resultsPage));
+      result = render(<Dashboard />, initialState([], 1));
     });
+    const rows = result?.container.querySelectorAll('tr');
+    expect(rows?.length).toBe(2);
+
     const itemsPerPageSelectorBtn = result?.container.querySelector(
       '[data-testid="itemsPerPageSelector"] > button'
     );
+    expect(mock.history.get).toHaveLength(1);
+    expect(mock.history.get[0].url).toContain('/notifications/sent');
     fireEvent.click(itemsPerPageSelectorBtn!);
     const itemsPerPageDropdown = await waitFor(() => screen.queryByRole('presentation'));
     expect(itemsPerPageDropdown).toBeInTheDocument();
     const itemsPerPageItem = within(itemsPerPageDropdown!).queryByText('50');
-    // reset mock dispatch function
-    mockDispatchFn.mockReset();
-    mockDispatchFn.mockClear();
-    fireEvent.click(itemsPerPageItem!);
-    await waitFor(() => {
-      expect(mockDispatchFn).toBeCalledTimes(1);
-      expect(mockDispatchFn).toBeCalledWith({
-        payload: { size: 50, page: 0 },
-        type: 'dashboardSlice/setPagination',
-      });
-    });
+
+    await waitFor(() => fireEvent.click(itemsPerPageItem!));
+    expect(mock.history.get).toHaveLength(2);
+    expect(mock.history.get[1].url).toContain('/notifications/sent');
+    const newRows = result?.container.querySelectorAll('tr');
+    expect(newRows?.length).toBe(3);
   });
 
   it('changes page', async () => {
+    mock = mockApi(
+      apiClient,
+      'GET',
+      NOTIFICATIONS_LIST({
+        startDate: formatToTimezoneString(tenYearsAgo),
+        endDate: formatToTimezoneString(getNextDay(today)),
+        size: 1,
+      }),
+      200,
+      undefined,
+      notificationsFromBe
+    );
+    mockApi(
```

Comment:
this mock is incorrect because we are not mocking the correct api. if we change page, the notification api is called with an extra parameter related to pagination